### PR TITLE
Emoji 15.1 alpha data

### PR DESCRIPTION
**Pull request by nedley**
Resolves [174-A61](https://www.unicode.org/cgi-bin/GetL2Ref.pl?174-A61) and #152.

### CODE REVIEW

**Comment by nedley** [comment]
Apparently I missed some changes for the removed duplicate files, working…

**Comment by nedley** [comment]
@markusicu It looks like `org.unicode.props.IndexUnicodeProperties.loadEnum` is still looking for emoji-data.txt in the emoji directory, can you fix it to check in ucd/emoji?

**Comment by markusicu** [comment]
> @markusicu It looks like `org.unicode.props.IndexUnicodeProperties.loadEnum` is still looking for emoji-data.txt in the emoji directory, can you fix it to use the one in ucd/emoji?

I will try to take a look, although that's really part of the work on issue #152, isn't it...?

File: unicodetools/data/emoji/dev/internal/emojiOrdering.txt [hunk 1]
```diff
@@ -0,0 +1,341 @@
+@@ Smileys_and_Emotion
+@ face-smiling
+😀 😃 😄 😁 😆 😅 🤣 😂
+🙂 🙃 🫠 😉 😊 😇
```
Comment (markusicu) [inline unicodetools/data/emoji/dev/internal/emojiOrdering.txt:4]: Why are you adding this file here?
Duplicate of unicodetools/src/main/resources/org/unicode/tools/emoji/emojiOrdering.txt ?

**Review by markusicu — COMMENTED** [review]
I hope that @macchiati takes a closer look at the `*Emoji*.java` files...

**Comment by nedley** [comment]
> I will try to take a look, although that's really part of the work on issue #152, isn't it...?

Technically yes, but if anything I would hope this is an easy fix if all other properties are also being read from UCD files.

File: unicodetools/data/emoji/dev/internal/emojiOrdering.txt (hunk 1 above)
Comment (nedley) [reply unicodetools/data/emoji/dev/internal/emojiOrdering.txt:4]: Please see my latest comment in #310.

**Comment by markusicu** [comment]
> @markusicu It looks like `org.unicode.props.IndexUnicodeProperties.loadEnum` is still looking for emoji-data.txt in the emoji directory, can you fix it to use the one in ucd/emoji?

I made a change, and running the UCD.Main tool locally works now. Pushed.

@macchiati could you please take a look at my changes in IndexUnicodeProperties.txt?

File: unicodetools/data/emoji/dev/internal/emojiOrdering.txt (hunk 1 above)
Comment (markusicu) [reply unicodetools/data/emoji/dev/internal/emojiOrdering.txt:4]: Looking at the history, Ned removed this file in PR #276, and in issue #283 we decided to keep it that way.
Please don't add it back again.

Comment (nedley) [reply unicodetools/data/emoji/dev/internal/emojiOrdering.txt:4]: OK, I will keep it in Generated for the time being.

Comment (markusicu) [reply unicodetools/data/emoji/dev/internal/emojiOrdering.txt:4]: Great, please remove this file from this PR.
“All checks have passed” but let me know if you need more help.

Comment (nedley) [reply unicodetools/data/emoji/dev/internal/emojiOrdering.txt]: [2f2a360](https://github.com/unicode-org/unicodetools/pull/397/commits/2f2a360e50334419c438dd4a4eeb448bce379597)

**Review by markusicu — DISMISSED** [review]
lgtm tnx -- please *squash-and-merge*

**Comment by nedley** [comment]
Sorry, one last fix to a comment that became stale after changing some code during development.

**Comment by macchiati** [comment]
@nedley It all looks great — nice changes! — but I have one question:

    static boolean altAppearance = true;

I would have to dig into the code to figure out what this does; could you just summarize why you changed it?

**Comment by nedley** [comment]
> I would have to dig into the code to figure out what this does; could you just summarize why you changed it?

Ah, yes. This was leftover from the last time I built the charts, it controls the alternating shading of columns on e.g. the full emoji list. For some reason I had to change it to match the previous round of charts and here we are.

**Review by macchiati — APPROVED** [review]
great work!

**Comment by markusicu** [comment]
@Ken-Whistler please close action item [174-A61](https://www.unicode.org/cgi-bin/GetL2Ref.pl?174-A61)

**Pull request merged**